addon/services/panel-actions.js

```diff
@@ -21,29 +21,35 @@ const State = EmberObject.extend({
   group: null
 });
 
-export default Service.extend({
-  _registry: EmberObject.create({
-    keys: A([]),
+const Registry = EmberObject.extend({
+  keys: computed(function() {
+    return A([]);
+  }),
 
-    unknownProperty: function(name) {
-      const state = State.create();
+  // probably not too safe, should only be used in tests
+  reset() {
+    this.get('keys')
+      .map(i => i) // copy, so we dont mess with binding/loops
+      .forEach((key) => {
+        delete this[key];
+      });
 
-      this.get('keys').addObject(name);
-      this.set(name, state);
+    this.get('keys').clear();
+  },
+});
 
-      return state;
-    },
+export default Service.extend({
+  _registry: computed(function() {
+    return Registry.create({
+      unknownProperty: function(name) {
+        const state = State.create();
 
-    // probably not too safe, should only be used in tests
-    reset() {
-      this.get('keys')
-        .map(i => i) // copy, so we dont mess with binding/loops
-        .forEach((key) => {
-          delete this[key];
-        });
+        this.get('keys').addObject(name);
+        this.set(name, state); // eslint-disable-line ember/no-side-effects
```

Comment:
There should be a meaningful, descriptive error thrown here if something isn't found.

Reply:
@NullVoxPopuli Thanks Preston for the review! This is how the component was implemented before this change.

I think this whole addon will get an overhaul soon once I find the time. I plan to Octanify it as well as separating the collapse logic from the UI.